lib/diplomat/rest_client.rb

```diff
@@ -304,6 +304,14 @@ def send_delete_request(connection, url, options, custom_params = nil)
       end
     end
 
+    # This method returns the correct Exception Classes depending on the Faraday version being installed
+    # see https://github.com/WeAreFarmGeek/diplomat/issues/227
+    #
+    # @return [Array<Class>] Faraday error classes that should be caught
+    def faraday_error_classes
+      Faraday.const_defined?(:ServerError) ? [Faraday::ClientError, Faraday::ServerError] : [Faraday::ClientError]
+    end
```

Comment:
Not entirely sure about the name/placement of this method, but is this like what you had in mind?

Reply:
Actually, i think it is bad to catch both ClientError and ServerError when ServerError exists, because it would thow a QueryAlreadyExists when a timeout occurs.

So I would use your code, but only return ServerError in such case, not ClientError, to avoid code calling this that this query exist while it is just that the agent did timeout for instance.

Reply:
Hmm, `Faraday::TimeoutError` is a client error in faraday < 1 and a server error in >= 1, even though the documentation claims it's a unified client error. As such, I decided to explicitly catch and reraise timeout errors.